src/helper/client.cpp

```diff
@@ -350,11 +348,20 @@ bool client::handle_command(network::request_exec::request &command)
 bool client::compute_client_status()
 {
     if (client_enabled_conf.has_value()) {
-        return client_enabled_conf.value();
+        request_enabled_ = client_enabled_conf.value();
+        return request_enabled_;
     }
 
-    return service_->get_service_config()->get_asm_enabled_status() ==
-           enable_asm_status::ENABLED;
+    if (service_ == nullptr) {
```

Comment:
A lack of service after `client_init` is a critical error / bug which should result in disconnecting the client, why include this check here?

Reply:
I added a few test ensuring that in the case that this scenario happen, we dont seg fault as it would do without that check. Current code does not allow to hit this case(only on testing). However, having this check here will prevent code changes in the future ending on unexpected scenarios with segfault

Reply:
Where would this segfault if you removed the check?

Reply:
Where would this segfault if you removed the check?

Reply:
Because of the next line. It accesses to the service with null value

Reply:
Another thing is `bool handle_command(network::request_init::request &);` is a public method of client. Nobody should call it before calling client init but it can happen so better to have this small check in place